NLP/word2vec-Pytorch/main.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from dataloader import loadData, prepareData, loadBatches
+from model import SkipGram
+from utils import subsampling, negativeSampling, getSimilarWords, getAnalogy, plotData
+import argparse
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument('--epochs', type = int, default = 50, help = "No of EPOCHS: default 5 ")
+parser.add_argument('--sampling_threshold', type = int, default = 1e-5, help = "Subsampling Threshold, default 1e-5 ")
+parser.add_argument('--min_freq', type = int, default = 5, help = "Minimum Frequency of words, default 5 ")
+parser.add_argument('--batch_size', type = int, default = 512, help = "Batch size, default 512")
+parser.add_argument('--window_size', type = int, default = 5, help = "Window size, default 5")
+parser.add_argument('--lr', type = int, default = 0.003, help = "Learning rate for generator optimizer,default 0.003 ")
+parser.add_argument('--embed_size', type = int, default = 300, help = "Embed size, default 300")
+parser.add_argument('--mode', type = str, default = "train", help = "Mode Type, default train")
+parser.add_argument('--N_words', type = int, default = 10, help = "Top N similar words, default 10")
+parser.add_argument('--word', type = str, default = "woman", help = "word, default word is woman")
+parser.add_argument('--word1', type = str, default = "king", help = "1st word, default king")
+parser.add_argument('--word2', type = str, default = "male", help = "2nd word, default male")
+parser.add_argument('--word3', type = str, default = "female", help = "3rd word, default female")
+args = parser.parse_args()
+
+
+with open('text8') as f:
+    text = f.read().split()
```

Comment:
`text8` file does not exist. If it is a data file, we recommend writing a code that automatically downloads the data.